File: src/main/kotlin/k/template_method/inheritance/CharDisplay.kt
/* (C)2022 */
package k.template_method.inheritance

class CharDisplay(val ch: Char) : AbstractDisplay() {

    override fun open() {
        // 開始文字列として"<<"を表示する
        print("<<")
    }

    override fun print() {
        // フィールドに保存しておいた文字を1回表示する
        print(ch)
    }

    override fun close() {
        // 終了文字列として">>"を表示する
        println(">>")
    }
}

File: src/main/scala/s/template_method/inheritance/CharDisplay.scala
package s.template_method.inheritance

final class CharDisplay(val ch: Char) extends AbstractDisplay {

  override def open(): Unit = {
    scala.Predef.print("<<")
  }

  override def print(): Unit = {
    scala.Predef.print(ch)
  }

  override def close(): Unit = {
    scala.Predef.println(">>")
  }

}
